app/src/main/java/com/example/atheneum/activities/MainActivity.java

```diff
@@ -103,6 +108,21 @@ public void onChanged(@Nullable User user) {
         } else {
             Log.w(TAG, "Shouldn't happen!");
         }
+
+        TestAggregateViewModelFactory factory = new TestAggregateViewModelFactory("sKrmipPG5KdImGQGk62efprGeWK2", "c27f0c04-ae7a-432b-9b88-5fd9b44585b0");
+        TestAggregateViewModel viewModel = ViewModelProviders.of(this, factory).get(TestAggregateViewModel.class);
+        LiveData<Pair<User, Book>> pairLiveData = viewModel.getBorrowerBookPairLiveData();
+        pairLiveData.observe(this, new Observer<Pair<User, Book>>() {
+            @Override
+            public void onChanged(@Nullable Pair<User, Book> userBookPair) {
+                if ((userBookPair != null) && (userBookPair.first != null) && (userBookPair.second != null)) {

```

Comment:
You have to check if the data is valid when you receive the data in the view. Instead of using a pair, I think a data class would be better in your case since you can have getters/setters for the data class elements and have a method to check if all the data is ready.